UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs

```diff
@@ -208,5 +209,56 @@ public string GenerateQueryForFss(List<Products> products)
             sb.Append(")");//// last main )
             return sb.ToString();
         }
+
+        public async Task<bool> DownloadBatchFiles(IEnumerable<string> uri, string downloadPath)
+        {
+            string payloadJson = string.Empty;
+            var accessToken = await authTokenProvider.GetManagedIdentityAuthAsync(fileShareServiceConfig.Value.ResourceId);
+            return await ProcessBatchFile(uri, downloadPath, payloadJson, accessToken);
+        }
+
+        private async Task<bool> ProcessBatchFile(IEnumerable<string> uri, string downloadPath, string payloadJson, string accessToken)
+        {
+            bool result = false;
+            foreach (var item in uri)
+            {
+                HttpResponseMessage httpResponse = await fileShareServiceClient.CallFileShareServiceApi(HttpMethod.Get, payloadJson, accessToken, item);
+                var fileName = item.Split("/").Last();
+                if (httpResponse.IsSuccessStatusCode)
+                {
+                    CheckCreateFolderPath(downloadPath);
+                    string path = Path.Combine(downloadPath, fileName);
+                    if (!File.Exists(path))
+                    {
+                        await CopyFileToFolder(httpResponse, path);
+                        result = true;
+                    }
+                }
+                else
+                {
+                    logger.LogInformation(EventIds.DownloadFileShareServiceNonOkResponse.ToEventId(), "File share service download end point with uri {RequestUri} responded with {StatusCode}", httpResponse.RequestMessage.RequestUri, httpResponse.StatusCode);
+                }
+            }
+            return result;
+        }
+
+        private static async Task CopyFileToFolder(HttpResponseMessage httpResponse, string path)
+        {
+            using (Stream stream = await httpResponse.Content.ReadAsStreamAsync())
```

Comment:
Code in this CopyFileToFolder can be simplified to below 3 lines
using Stream stream = await httpResponse.Content.ReadAsStreamAsync();
using FileStream outputFileStream = new FileStream(path, FileMode.Create, FileAccess.ReadWrite); stream.CopyTo(outputFileStream);

Reply:
modified accordingly